Library/Homebrew/extend/os/mac/system_config.rb

```diff
@@ -45,12 +45,25 @@ def core_tap_config(out = $stdout)
           dump_tap_config(CoreCaskTap.instance, out)
         end
 
+        sig { returns(T.nilable(String)) }
+        def metal_toolchain
+          @metal_toolchain ||= T.let(if MacOS::Xcode.installed? || MacOS::CLT.installed?
+                                       result = SystemCommand.run("xcrun", args: ["metal", "-v"],
+                                       print_stderr: false, print_stdout: false)
+                                       "present" if result.success?
```

Comment:
Maybe the version number would be useful here? It might currently be tied to the Xcode version, but that might not always be the case.

Reply:
I've yet to find a way to get that version outside of the Xcode UI. The command in this PR returns this:

```
Apple metal version 32023.830 (metalfe-32023.830.2)
```